eli5/sklearn/utils.py

```diff
@@ -29,28 +37,49 @@ def get_feature_names(clf, vec=None, bias_name='<BIAS>', feature_names=None):
         if vec and hasattr(vec, 'get_feature_names'):
             feature_names = vec.get_feature_names()
         else:
-            num_features = clf.coef_[0].shape[0]
+            num_features = clf.coef_.shape[-1]
             feature_names = ["x%d" % i for i in range(num_features)]
-        if bias_name is not None and has_intercept(clf):
-            feature_names += [bias_name]
     else:
-        num_features = clf.coef_[0].shape[0] + int(has_intercept(clf))
+        num_features = clf.coef_.shape[-1]
         if len(feature_names) != num_features:
-            raise ValueError("feature_names has a wrong lenght: "
+            raise ValueError("feature_names has a wrong length: "
                              "expected=%d, got=%d" % (num_features,
                                                       len(feature_names)))
+    if bias_name is not None and has_intercept(clf):
+        feature_names += [bias_name]
     return np.array(feature_names)
 
 
+def get_target_names(clf):
+    """
+    Return a vector of target names: y if the is only one target,
+    and y0, ... if there are multiple targets.
+    """
+    if len(clf.coef_.shape) == 1:
+        target_names = ['y']
+    else:
+        num_targets, _ = clf.coef_.shape
+        target_names = ['y%d' % i for i in range(num_targets)]
+    return np.array(target_names)
+
+
 def get_coef(clf, label_id):
     """
     Return a vector of coefficients for a given label,
     including bias feature.
     """
-    coef = clf.coef_[label_id]  # multiclass case, also works for binary
+    if len(clf.coef_.shape) == 2:
+        coef = clf.coef_[label_id]  # multiclass case, also works for binary
```

Comment:
Right, SGDRegressor stores coefficients as a one-dim array, and binary does not make sense here, I'll fix comments